modules/library/main/src/main/java/org/geotools/filter/visitor/SimplifyingFilterVisitor.java

```diff
@@ -418,7 +423,13 @@ public Object visit(Not filter, Object extraData) {
                 PropertyIsBetween pb = (PropertyIsBetween) simplified;
                 Filter lt = ff.less(pb.getExpression(), pb.getLowerBoundary());
                 Filter gt = ff.greater(pb.getExpression(), pb.getUpperBoundary());
-                return ff.or(lt, gt);
+                PropertyName pn = (PropertyName) pb.getExpression();
+                String pName = pn.getPropertyName();
+                Filter nullFilter = null;
+                if(isNillable(pName)) {
+                	nullFilter = ff.isNull(pb.getExpression());
+                }
+                return ff.or(Stream.of(lt,gt,nullFilter).filter(Objects::nonNull).collect(Collectors.toList()));
```

Comment:
If I'm not mistaken the propose of the expression ``Stream.of(lt,gt,nullFilter).filter(Objects::nonNull).collect(Collectors.toList()) `` is to get rid of ``nullFilter`` if it's ``NULL``, I think ti would be more efficient and clear to do something like this instead:
```java
if (isNillable(pName)) {
    // the property may have a NULL value, we need to vouch for it
    nullFilter = ff.isNull(pb.getExpression());
    return ff.or(lt, gt, nullFilter);
}
// the property value is not nullable
return ff.or(lt, gt);
```

Reply:
Not possible to use ff.or(lt, gt, nullFilter); because it accepts 2 parameters or an ArrayList (for more than 2 filters).

Reply:
You're correct, my mistake. In that case I would suggest to separate the concerns; first you build the list of filters then you build the filter. Something like this:
```java
List<Filter> orFilters = new ArrayList<>();
orFilters.add(lt);
orFilters.add(gt);
if (isNillable(pName)) {
    // the property may have a NULL value, we need to vouch for it
    orFilters.add(pb.getExpression());
}
return ff.or(orFilters);
```

Reply:
Dear @nmco, I updated the file according to your suggestion.